File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n = mat.length;
        int result = 0;
        for (int i = 0; i < n; i++) {
            result += mat[i][i];
            result += mat[i][n - i - 1];
        }
        if (n % 2 == 0) {
            return result;
        }
        else {
            return result - mat[n / 2][n / 2];
        }
    }
}

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        dim = len(mat)
        result = 0
        
        for i in range(dim):
            result += mat[i][i]
            result += mat[i][dim - i - 1]
        if dim % 2 == 1:
            return result -mat[dim//2][dim//2]
        return result
            
        